From d651ca00b9162eb20eb11e0fb40b80db9face395 Mon Sep 17 00:00:00 2001
From: Alaa Hleihel <ahleihel@redhat.com>
Date: Mon, 1 Jun 2020 15:41:01 -0400
Subject: [PATCH 305/312] [netdrv] net/mlx5e: Remove warning "devices are not
 on same switch HW"

Message-id: <20200601154102.25980-39-ahleihel@redhat.com>
Patchwork-id: 315743
Patchwork-instance: patchwork
O-Subject: [RHEL8.3 BZ 1842258 38/39] net/mlx5e: Remove warning "devices are not on same switch HW"
Bugzilla: 1842258
RH-Acked-by: Honggang Li <honli@redhat.com>
RH-Acked-by: Kamal Heib <kheib@redhat.com>
RH-Acked-by: Marcelo Leitner <mleitner@redhat.com>
RH-Acked-by: Jarod Wilson <jarod@redhat.com>

Bugzilla: http://bugzilla.redhat.com/1842258
Upstream: v5.7

commit 20300aafa7a2719f71d50f97a8846459d9869b75
Author: Maor Dickman <maord@mellanox.com>
Date:   Sun May 24 09:45:44 2020 +0300

    net/mlx5e: Remove warning "devices are not on same switch HW"

    On tunnel decap rule insertion, the indirect mechanism will attempt to
    offload the rule on all uplink representors which will trigger the
    "devices are not on same switch HW, can't offload forwarding" message
    for the uplink which isn't on the same switch HW as the VF representor.

    The above flow is valid and shouldn't cause warning message,
    fix by removing the warning and only report this flow using extack.

    Fixes: 321348475d54 ("net/mlx5e: Fix allowed tc redirect merged eswitch offload cases")
    Signed-off-by: Maor Dickman <maord@mellanox.com>
    Reviewed-by: Roi Dayan <roid@mellanox.com>
    Signed-off-by: Saeed Mahameed <saeedm@mellanox.com>

Signed-off-by: Alaa Hleihel <ahleihel@redhat.com>
Signed-off-by: Frantisek Hrbata <fhrbata@redhat.com>
---
 drivers/net/ethernet/mellanox/mlx5/core/en_tc.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/drivers/net/ethernet/mellanox/mlx5/core/en_tc.c b/drivers/net/ethernet/mellanox/mlx5/core/en_tc.c
index 2df19165a78a..c5966589625f 100644
--- a/drivers/net/ethernet/mellanox/mlx5/core/en_tc.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/en_tc.c
@@ -3830,10 +3830,6 @@ static int parse_tc_fdb_actions(struct mlx5e_priv *priv,
 				if (!mlx5e_is_valid_eswitch_fwd_dev(priv, out_dev)) {
 					NL_SET_ERR_MSG_MOD(extack,
 							   "devices are not on same switch HW, can't offload forwarding");
-					netdev_warn(priv->netdev,
-						    "devices %s %s not on same switch HW, can't offload forwarding\n",
-						    priv->netdev->name,
-						    out_dev->name);
 					return -EOPNOTSUPP;
 				}
 
-- 
2.13.6

